AppCenter/AppCenter/Internals/Storage/MSLogDBStorage.m

```diff
@@ -277,9 +277,15 @@ - (void)deleteLogsWithBatchId:(NSString *)batchId groupId:(NSString *)groupId {
 
     // Deserialize target token.
     NSString *encryptedToken = row[self.targetTokenColumnIndex];
-    if (![encryptedToken isKindOfClass:[NSNull class]]) {
+    if (![encryptedToken isKindOfClass:[NSNull class]] && (encryptedToken.length > 0)) {
       NSString *targetToken = [self.targetTokenEncrypter decryptString:encryptedToken];
-      [log addTransmissionTargetToken:targetToken];
+      if (targetToken) {
+        [log addTransmissionTargetToken:targetToken];
+      } else {
+        MSLogWarning([MSAppCenter logTag], @"Failed to decrypt the target token.");
+      }
+    } else {
+      MSLogVerbose([MSAppCenter logTag], @"Encrypted token was NSNull or an empty string (this is not a 1DS log).");
```

Comment:
`1DS` -> `CS`.

Reply:
We can just remove the parenthesis I think. It's hypothesis.

Reply:
That works too.

Reply:
I revised the `if` logic a bit to only print an error when it's an actual error (empty string).